### PR TITLE
feat: rollup-dev-server for CT

### PR DESCRIPTION
**Pull request by lmiller1990**
Add a `rollup-dev-server` package based on [Nollup](https://github.com/PepsRyuu/nollup).

Everything seems to work as expected.  Just `yarn cy:open` and it works. 

I had to do some somewhat dirty things to bundle everything correctly. Cypress will request the spec by original name (eg `blah.spec.ts`) but Nollup will always output `.js`. You should be able to configure this but it seems not to work 🤔 not sure why, seems some Rollup features are not ported to Nollup.

To inject the HMR I had to write a sneaky little transformer, works great.

### CODE REVIEW

File: npm/rollup-dev-server/cypress.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "experimentalComponentTesting": true,
```
Comment (dmtrKovalenko) [inline npm/rollup-dev-server/cypress.json:2]: This is not needed I think

**Review by dmtrKovalenko — COMMENTED** [review]
Not sure where `dev-server:compile:error` and `dev-server:compile:success` events implemented. They are required to actually work with dev-server, right?

File: npm/rollup-dev-server/cypress.json (hunk 1 above)
Comment (lmiller1990) [reply npm/rollup-dev-server/cypress.json]: fixed

File: npm/rollup-dev-server/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  "files": [
+    "dist"
+  ],
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/cypress-io/cypress.git"
+  },
+  "homepage": "https://github.com/cypress-io/cypress/tree/master/npm/vite-dev-server#readme",
```
Comment (JessicaSachs) [inline npm/rollup-dev-server/package.json:35]:
```suggestion
  "homepage": "https://github.com/cypress-io/cypress/tree/master/npm/rollup-dev-server#readme",
```

File: npm/rollup-dev-server/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export interface ResolvedDevServerConfig {
+  port: number
+  server: Server
+}
+
+export async function startDevServer (startDevServerArgs: StartDevServer): Promise<ResolvedDevServerConfig> {
+  const rollupDevServer = await createDevServer(startDevServerArgs)
+
+  return new Promise(async (resolve) => {
+    const port = 3000
```
Comment (JessicaSachs) [inline npm/rollup-dev-server/src/index.ts:30]: Can we do a random port...? Should be able to pass in `port: 0` to do this.

File: npm/rollup-dev-server/src/startServer.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  })
+
+  const app = express()
+  const server = http.createServer(app)
+  const contentBase = resolve(__dirname, devServerOptions.options.config.projectRoot)
+
+  const nollup = NollupDevMiddleware(app, config, {
+    contentBase,
+    port: 3000,
```
Comment (JessicaSachs) [inline npm/rollup-dev-server/src/startServer.ts:52]: Repeated port. Need to pass it through so everything stays in sync.

**Review by JessicaSachs — CHANGES_REQUESTED** [review]
Notes:
* Need to do a random port instead of hardcoding the number from the 2 places it's used (block)
* Specs are mostly irrelevant (nit?)
* Missing CI step (block)
* 
~~

### RE: Specs

The specs here are kinda irrelevant. They were written for Vite which has particular behaviors wrt non-ESM modules and module optimization.

The support spec is probably relevant though. I suppose this support file might be one of those cases where we hook this up into the playground. Yknow? We should always be able to load the support file... that's a common use case.

File: npm/rollup-dev-server/src/startServer.ts (hunk 4 above)
Comment (lmiller1990) [reply npm/rollup-dev-server/src/startServer.ts]: done

File: npm/rollup-dev-server/src/index.ts (hunk 3 above)
Comment (lmiller1990) [reply npm/rollup-dev-server/src/index.ts]: port: 0 does not work for nollup apparently so I just made a random number instead.

File: npm/rollup-dev-server/package.json (hunk 2 above)
Comment (lmiller1990) [reply npm/rollup-dev-server/package.json]: done

**Review by JessicaSachs — APPROVED** [review]

**Pull request merged**